src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneService.cs

```diff
@@ -54,16 +59,18 @@ public async ValueTask UpdateUserTimeZoneAsync(IUser user)
         return;
     }
 
-    public async ValueTask<string> GetCurrentUserTimeZoneIdAsync()
+    /// <inheritdoc/>
+    private async ValueTask<string> GetTimeZoneIdAsync()
```

Comment:
This method should accept username as a parameter and not use HttpContext.User.

Reply:
We could pass `IUser` instead